velox/exec/tests/RowContainerTest.cpp

```diff
@@ -187,6 +190,60 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     testRowNumbers();
   }
 
+  void testRowNumbersSpecialCases(
+      RowContainer& container,
+      const std::vector<char*>& rows,
+      int column,
+      const VectorPtr& expected) {
+    auto size = rows.size();
+    auto testRepeatedRowNumber = [&]() {
+      // Copy the first row to all positions in the result vector
+      std::vector<vector_size_t> rowNumbers;
+      rowNumbers.resize(size);
+      for (int i = 0; i < size; i++) {
```

Comment:
You can replace this loop using an std::vector constructor that takes number of values and the initial value.

```
std::vector<vector_size_t> rowNumbers(size, 0);
```